src/CMakeLists.txt

```diff
@@ -3,24 +3,31 @@ project (AtomVM)
 
 add_subdirectory(libAtomVM)
 
+if((${CMAKE_SYSTEM_NAME} STREQUAL "Darwin") OR 
+   (${CMAKE_SYSTEM_NAME} STREQUAL "Linux") OR 
+   (${CMAKE_SYSTEM_NAME} STREQUAL "FreeBSD"))
+    add_subdirectory(platforms/generic_unix)
+else()
+    message(FATAL_ERROR "Unsupported platform: ${CMAKE_SYSTEM_NAME}")
+endif()
+
+if(${CMAKE_SYSTEM_NAME} STREQUAL "Linux")
+    set(RT_LIB "rt")
+endif()
+
+set(
+    PLATFORM_LIB_SUFFIX
+    ${CMAKE_SYSTEM_NAME}-${CMAKE_SYSTEM_PROCESSOR}
+)
+
 if(CMAKE_COMPILER_IS_GNUCC)
     set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wall -pedantic -Wextra -ggdb")
 endif()
 
 include_directories(${CMAKE_CURRENT_BINARY_DIR} libAtomVM/)
 
-if(${CMAKE_SYSTEM_NAME} STREQUAL "Darwin")
-    set(PLATFORM_DIR "generic_unix")
-elseif(${CMAKE_SYSTEM_NAME} STREQUAL "Linux")
-    set(PLATFORM_DIR "generic_unix")
-elseif(${CMAKE_SYSTEM_NAME} STREQUAL "FreeBSD")
-    set(PLATFORM_DIR "generic_unix")
-else()
-    message(FATAL_ERROR "Unsupported platform: ${CMAKE_SYSTEM_NAME}")
-endif()
-
-add_executable(AtomVM main.c platforms/${PLATFORM_DIR}/sys.c platforms/${PLATFORM_DIR}/mapped_file.c platforms/${PLATFORM_DIR}/udpdriver.c)
-target_link_libraries(AtomVM libAtomVM)
+add_executable(AtomVM main.c)
+target_link_libraries(AtomVM libAtomVM libAtomVM${PLATFORM_LIB_SUFFIX} ${RT_LIB})
```

Comment:
I'm not really sure yet about this rt change, it looks useless on most modern linux distros, I need to check deeper why is needed on Centos and if there is any cleaner way to do this.

Reply:
This is the point: https://flameeyes.blog/2012/12/26/glibc-2-17-what-s-going-to-be-a-trouble/
I will check how to handle this :)

Reply:
I will back out this change.

Reply:
If we want to support older systems like Centos 6 we just need this:
https://stackoverflow.com/questions/13747505/why-doesnt-check-function-exists-find-clock-gettime-in-cmake

Reply:
Applied the above recommendation and amended commit.